cli/lsp/diagnostics.rs

```diff
@@ -351,7 +351,7 @@ async fn generate_lint_diagnostics(
 }
 
 async fn generate_ts_diagnostics(
-  snapshot: &language_server::StateSnapshot,
+  snapshot: Arc<language_server::StateSnapshot>,
```

Comment:
Another change in this PR is to wrap `StateSnapshot` in an `Arc` to (1) prevent someone thinking it would be ok to mutate the values in it and (2) documents is more expensive to clone now so this makes cloning a state snapshot very cheap.